### PR TITLE
boards: st: stm32h735g_disco: enable USB device support

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Enable USB device support on the stm32h735g_disco board.

### CODE REVIEW

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Comment by aescolar** [comment]
@henrikbrixandersen , @erwango would you be so kind as to indicate to which release you would like this PR to go?
Thanks in advance.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen , @erwango would you be so kind as to indicate to which release you would like this PR to go? Thanks in advance.

I believe it is isolated enough to be considered for v3.7.0, but I'll leave that up to @erwango as the platform maintainer.

**Pull request merged**
